pkg/sync/sync_context.go

```diff
@@ -152,10 +152,11 @@ func WithResourceModificationChecker(enabled bool, diffResults *diff.DiffResultL
 }
 
 // WithNamespaceCreation will create non-exist namespace
-func WithNamespaceCreation(createNamespace bool, namespaceModifier func(*unstructured.Unstructured) bool) SyncOpt {
+func WithNamespaceCreation(createNamespace bool, namespaceModifier func(*unstructured.Unstructured) bool, namespaceCreator func(*unstructured.Unstructured) bool) SyncOpt {
 	return func(ctx *syncContext) {
 		ctx.createNamespace = createNamespace
 		ctx.namespaceModifier = namespaceModifier
+		ctx.namespaceCreator = namespaceCreator
```

Comment:
Do you consider renaming `namespaceCreator`  to `createNamespaceMetadata ` ?  namespaceCreator is very similar to  `createNamespace`

Reply:
it is hook, inside this hook can be another logic, not only metadata one